File: examples/example-63/ruby/features/pages/SimpleSearchPage.rb
require 'selenium-webdriver'

class SimpleSearchPage
    def initialize(driver, base_url)
        @driver = driver
        @url = "#{base_url}/index.html"
        @search_box_locator = { id: 'search_box' }
        @results_locator = { id: 'results' }
    end

    def open
        @driver.navigate.to @url
    end

    def type_search(query)
        wait = Selenium::WebDriver::Wait.new(timeout: 10)
        wait.until { @driver.find_element(@search_box_locator).displayed? }
        search_box = @driver.find_element(@search_box_locator)
        search_box.clear
        search_box.send_keys(query)
    end

    def submit_search
        wait = Selenium::WebDriver::Wait.new(timeout: 10)
        wait.until { @driver.find_element(@search_box_locator).displayed? }
        search_box = @driver.find_element(@search_box_locator)
        search_box.submit
    end

    def results_text
        wait = Selenium::WebDriver::Wait.new(timeout: 10)
        wait.until { @driver.find_element(@results_locator).displayed? }
        results_area = @driver.find_element(@results_locator)
        results_area.text
    end
end

File: examples/example-64/js/features/pages/SimpleSearchPage.js
const { By, until, Key } = require('selenium-webdriver');

class SimpleSearchPage {
    constructor(driver, baseUrl) {
        this.driver = driver;
        this.url = `${baseUrl}/index.html`;
        this.searchBoxLocator = By.id('search_box');
        this.resultsLocator = By.id('results');
    }

    async open() {
        await this.driver.get(this.url);
    }

    async typeSearch(query) {
        let searchBox = await this.driver.wait(
            until.elementLocated(this.searchBoxLocator), 10000);
        await searchBox.clear();
        await searchBox.sendKeys(query);
    }

    async submitSearch() {
        let searchBox = await this.driver.findElement(this.searchBoxLocator);
        await searchBox.sendKeys(Key.RETURN);
    }

    async results() {
        let resultsArea = await this.driver.wait(
            until.elementLocated(this.resultsLocator), 10000);
        return await resultsArea.getText();
    }
}

module.exports = SimpleSearchPage;